guides/lagom-dcos-marathon-deploy-microservices/src/main/paradox/index.md

```diff
@@ -0,0 +1,338 @@
+    {  "^chirpservice$"     = "_chirp-impl-chirper._tcp.marathon.mesos" },
+    {  "^friendservice$"    = "_friend-impl-chirper._tcp.marathon.mesos" },
+    {  "^.*$"               = "_$0._tcp.marathon.mesos" }
+  ]
+
+  srv-translators = [
+    { "^_(.+)-chirper[.]_tcp[.]marathon.mesos$" = "_$1-chirper._http.marathon.mesos" },
+    { "^.*$"                                    = "$0" }
+  ]
+}
+```
+
+_Refer to the various `marathon-resources/platform.conf` files in the Chirper repository for more details._
+
+#### Akka Cluster Bootstrapping
+
+Lagom's persistence APIs rely on Akka Persistence and thus Akka Cluster. This means that great care must be taken
+when deploying these applications to ensure that they are bootstrapped correctly and do not form separate "islands" of
+1-node clusters. To do this, Chirper is configured to use [ConstructR](https://github.com/hseeberger/constructr) and
+[ConstructR-ZooKeeper](https://github.com/typesafehub/constructr-zookeeper) to bootstrap the cluster. For your own
+project, you'll need to add both of these libraries to the project's dependencies and configure as necessary. Chirper
+uses the following configuration in its `application.conf` files:
+
+```
+akka.extensions = [de.heikoseeberger.constructr.ConstructrExtension]
+
+constructr.coordination.nodes = ${?CONSTRUCTR_COORDINATION_NODES}
+```
+
+The Marathon configuration is responsible for specifying the value of `CONSTRUCTR_COORDINATION_NODES`. Chirper
+is configured to use one instance. If you have multiple DC/OS master nodes, be sure to specify all of the ZooKeeper
+nodes in a comma-separated format.
+
+```
+# Single node
+
+"CONSTRUCTR_COORDINATION_NODES": "zk-1.zk:2181/mesos"
+
+```
+
+```
+# Multiple nodes
+
+"CONSTRUCTR_COORDINATION_NODES": "zk-1.zk:2181/mesos,zk-2.zk:2181/mesos,zk-3.zk:2181/mesos"
+```
+
+_Refer to the various `marathon-resources/platform.conf` and `deploy/marathon/resources/chirper.json` files in the 
+Chirper repository for more details. Be sure to consult the README for ConstructR and ConstructR-ZooKeeper as well._
+
+#### Dynamic Proxying & Ingress Routing
+
+This guide uses [Marathon-LB](https://github.com/mesosphere/marathon-lb) to provide dynamic proxying and ingress
+routing. This ensures that, for example, a request to `/api/chirpservice` is routed to `chirp-impl` while a request to
+`/api/users` is routed to `friend-impl`. To do this, various labels must be set in the Marathon configuration. For 
+example, `friend-impl` uses the following:
+
+```
+"labels": {
+  "HAPROXY_GROUP": "external",
+  "HAPROXY_0_VHOST": "p1.dcos",
+  "HAPROXY_0_PATH": "/api/users"
+}
+```
+
+_Refer to `deploy/marathon/resources/chirper.json` for more information on the Marathon-LB configuration._
+
+## Local Cluster
+
+The [DC/OS Vagrant](https://github.com/dcos/dcos-vagrant) project can be used to provision a local cluster on your
+own machine using a number of virtual machines. Consult its README for more information.
```

Comment:
We might also want to note memory requirements. The first time I tried to start up that config (on my 16GB laptop) it caused the system to crash. After reboot and ensuring that I closed other RAM-hungry apps I got it to work, but it is bursting at the seams a bit.

Reply:
Even this configuration might be insufficient. I am seeing that marathon-lb is not being deployed due to insufficient memory on the public agent. On the other hand, the private agents have spare capacity. Perhaps we should supply our own config file.

Reply:
Yikes - I'm a bit out of touch with my 32GB  :blush: -- there I'm using the `VagrantConfig-1m-3a-1p.yaml` config (`cp` it to `VagrantConfig.yaml` to "activate" it)

I'll try this with a 16GB machine tomorrow and update the guide to address any pain points and expand on the DC/OS Vagrant configuration.

Reply:
Although I will say the amount of ram it requires in general is painful. Even makes Minikube look light weight :stuck_out_tongue:

Reply:
No easy way so far to get this to work on 16GB because of Cassandra -- which the Universe packaging eats up 16GB minimum out of the box. I guess I'll try running a single node Docker image in Marathon and offering both choices in the guide.

Reply:
It's probably not that important to get it running in a 16GB environment... it's enough to just warn people away from trying.